### ISSUE TITLE
Add customized timeout value to wrapper

### ISSUE THREAD

**Issue by rickmurt** [onfleet/node-onfleet#24]
**Describe the bug**
I'm having an issue in which if the `notes` field in the payload passed to `onfleet.tasks.create` contains a newline (\n) it _**sometimes**_ causes a server timeout. There is no error code returned, my 30 second max timeout on my server just triggers and the task creation fails. I can try to pass the same exact payload along after restarting the server, and it works perfectly fine, receiving a response in ~150ms. This _only_ happens if there is a newline character in the `notes` field, and only happens intermittently. I went without a newline character in the notes field for about 4 months, and never once received a timeout. I then added newline characters, and was receiving timeouts every few hours. I reverted back to removing the newline character, and I've now gone 12 hours (hundreds of tasks) without an error.

I understand this is bizarre, and intermittent issues are incredibly difficult to troubleshoot, but it was the only change to my code which was working beautifully for months. Reverting it immediately fixed the problem.

**To Reproduce**
1. Send this semi-anonymized payload and have server timeout every few hours 
```
{"destination":{"address":{"unparsed":"123 My Street, City, TX 78738"},"notes":"Please place inside gate at bottom of the driveway. Right side opens out. "},"recipients":[{"name":"First Last","phone":"5555551234"}],"completeBefore":1597363235850,"completeAfter":1597330835850,"notes":"4x Line Item 1, 8 oz\n2x Line Item 2, 4 oz\n2x Line Item 3, 4 oz\n2x Line Item 4, 4 oz \n  \n123 My Street , City, TX 78738\n  \nhttps://mysubdomain.myshopify.com/admin/orders/1234567.","quantity":10,"container":{"type":"TEAM","team":"MY_TEAM_ID_HERE"}}
```
2. Change it to the following format without \n newline characters and never have the server timeout
```
{"destination":{"address":{"unparsed":"123 My Street, City, TX 78738"},"notes":"Please place inside gate at bottom of the driveway. Right side opens out. "},"recipients":[{"name":"First Last","phone":"5555551234"}],"completeBefore":1597363235850,"completeAfter":1597330835850,"notes":"https://mysubdomain.myshopify.com/admin/orders/1234567; 4x Line Item 1, 8 oz; 2x Line Item 2, 4 oz; 2x Line Item 3, 4 oz; 2x Line Item 4, 4 oz; Unparsed Address: My Street , City, TX 78738","quantity":10,"container":{"type":"TEAM","team":"MY_TEAM_ID_HERE"}}
```

**Additional context**
I tried both versions `1.0.6` and `1.1.0` and was able to produce with both. I don't have any applicable IDs as the request timed out, and never made it to the `.catch` of my `onfleet.tasks.create` call.

**Comment by jamesliupenn** [comment]
@rickmurt have you encountered the issue since you last reported? It is definitely bizarre as the issue seems intermittent and only happens when `\n` is present in the notes field. The notes field can definitely handle newline characters so I'm a little lost here.

Also, since you don't have any request ID, it means that the HTTP request never reached Onfleet's servers - making it even harder for our Support team to troubleshoot..... 🔍 

Does anyone else have the same issue in the community?

**Comment by rickmurt** [comment]
@jamesliupenn as it turns out, it wasn't the newline character. I fixed this by going back to version 1.0.6. I thought I had tried that previously but my yarn.lock file was incorrect. I fixed this a long time ago so I don't recall the exact details but I think after tracing through the node-onfleet code, I figured that the reason 1.0.6 worked and 1.1.0 (or 1.0.7 - 1.1.0) didn't is because 1.1.0 had a timeout value hardcoded that was longer than my heroku servers max timeout (30 seconds). I think I traced it back to this commit: https://github.com/onfleet/node-onfleet/commit/35aacac3c6d472343279e5e5d043f9f74c86db39

I think the bumping of the max timeout from 10 seconds to 70 was the culprit here because it put me over 30 seconds. If we could pass in our own timeout value, then I could make it something like 20 seconds. In the meantime, I've continued to use 1.0.6

**Comment by jamesliupenn** [comment]
@rickmurt I'm going to put in an optional field for you to pass in your custom timeout value.

So instead of doing `new Onfleet(api_key)`, you can instead do `new Onfleet(api_key, 20000)` to set a customized timeout value in milliseconds. I believe this would resolve your issue, can you verify?

**Comment by jamesliupenn** [comment]
@rickmurt can you verify the PR and confirm that the added customize timeout value resolves your issue: https://github.com/onfleet/node-onfleet/pull/32

**Comment by rickmurt** [comment]
> @rickmurt can you verify the PR and confirm that the added customize timeout value resolves your issue: #32

@jamesliupenn that's perfect. Thank you!

**Comment by jamesliupenn** [comment]
This is now available on github and npm @rickmurt
